src/PayerRegistry.sol

```diff
@@ -0,0 +1,530 @@
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function setMinimumDeposit(uint96 newMinimumDeposit_) external onlyAdmin {
+        _setMinimumDeposit(newMinimumDeposit_);
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function setWithdrawLockPeriod(uint32 newWithdrawLockPeriod_) external onlyAdmin {
+        _setWithdrawLockPeriod(newWithdrawLockPeriod_);
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function setMinActiveBalance(int104 newMinActiveBalance_) external onlyAdmin {
+        _setMinActiveBalance(newMinActiveBalance_);
+    }
+
+    /* ============ View/Pure Functions ============ */
+
+    /// @inheritdoc IPayerRegistry
+    function admin() external view returns (address admin_) {
+        return _getPayerRegistryStorage().admin;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function settler() external view returns (address settler_) {
+        return _getPayerRegistryStorage().settler;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function feeDistributor() external view returns (address feeDistributor_) {
+        return _getPayerRegistryStorage().feeDistributor;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function totalDeposits() public view returns (int104 totalDeposits_) {
+        return _getPayerRegistryStorage().totalDeposits;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function totalDebt() public view returns (uint96 totalDebt_) {
+        return _getPayerRegistryStorage().totalDebt;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function totalWithdrawable() external view returns (uint96 totalWithdrawable_) {
+        return _getTotalWithdrawable(totalDeposits(), totalDebt());
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function minimumDeposit() external view returns (uint96 minimumDeposit_) {
+        return _getPayerRegistryStorage().minimumDeposit;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function withdrawLockPeriod() external view returns (uint32 withdrawLockPeriod_) {
+        return _getPayerRegistryStorage().withdrawLockPeriod;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function minActiveBalance() public view returns (int104 minActiveBalance_) {
+        return _getPayerRegistryPeripheryStorage().minActiveBalance;
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function activePayersCount() external view returns (uint256 count_) {
+        return _activePayers().length();
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function activePayers() external view returns (address[] memory activePayers_) {
+        return _activePayers().values();
+    }
+
+    /// @inheritdoc IPayerRegistry
+    function getActivePayers(
+        uint256 fromIndex_,
+        uint256 count_
+    ) external view returns (address[] memory activePayers_) {
+        activePayers_ = new address[](count_);
+
+        for (uint256 index_; index_ < count_; ++index_) {
+            activePayers_[index_] = _activePayers().at(fromIndex_ + index_);
+        }
+    }
```

Comment:
The idea behind `getActivePayers` is/was to help nodes bootstrap as fast as possible and ideally it would include each payer's balance, so the node doesn't have to call the blockchain again and instead just switches to filtering or subscribing from the latest block.

Reply:
Good point. I will create a `function getActivePayersAndBalances(uint256 fromIndex_, uint256 count_) external view returns (address[] memory activePayers_, int104[] memory balances_)'`

Reply:
Since we don't have the concept of an active payer anymore, `GetPayersWithBalance`? Or similar.

Reply:
If we drop `minActiveBalance`, which I am making an argument in other comments not to, then we would have `getHealthyPayers` and `getHealthyPayersAndBalances`.

Reply:
Periphery removed in latest commit, as per agreement on call.